sre-recipes/recipes/encoding_recipe/encoding_recipe.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- coding: utf-8 -*-
+"""
+This module contains the implementation of recipe 1
+"""
+
+import logging
+import subprocess
+from recipe import Recipe
+
+
+class EncodingRecipe(Recipe):
+    """
+    This class implements recipe 1, which purposefully
+    spits errors from the Email Service.
+    """
+    name = "encoding_recipe"
+
+    def get_name(self):
+        return self.name
+        
+    def deploy_state(self, state):
+        """
+        Sets an environment variable ENCODE_EMAIL to given state
+        and updates the state accordingly
+        """
+        state_str = str(state).lower()
+        set_env_command = f"kubectl set env deployment/emailservice ENCODE_EMAIL={state_str}"
+        get_pod_command = """kubectl get pod -l app=emailservice -o \
+            jsonpath=\"{.items[0].metadata.name}\""""
+        logging.info('Setting env variable: %s', set_env_command)
+        logging.info('Getting pod: %s', get_pod_command)
+
+        Recipe._run_command(set_env_command)
+        service, error = Recipe._run_command(get_pod_command)
+        service = service.decode("utf-8").replace('"', '')
+        if not service:
+            print('No service found. Could not deploy state.')
+            logging.error('No service found. Could not deploy state.')
+            exit(1)
+        delete_pod_command = f"kubectl delete pod {service}"
+        logging.info('Deleting pod: %s', delete_pod_command)
+        Recipe._run_command(delete_pod_command)
```

Comment:
Gotcha - I added a wait command. I think it's okay for users to see that only one service is reset since hopefully they're using the recipes to learn Cloud Ops. Otherwise they could just look into the recipe code directory anyways.

Reply:
I'd recommend making the wait on the emailservice, since that's the one that's reset. 

And yeah, that makes sense, just thought I would point out that `kubectl get pods` gives a bit of a spoiler if they wanted to hunt it down themselves haha

Reply:
Fixed!